### PR TITLE
Honey-13: Added documentation

### PR DESCRIPTION
**Pull request by MarsianMan**
Examples for manually setting Honeycomb's trace header using popular Java HTTP client frameworks.

### CODE REVIEW

File: beeline-core/src/main/java/io/honeycomb/beeline/tracing/README.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+request.addHeader("x-honeycomb-trace", MessageFormat.format("1;trace_id={0},parent_id={1}", span.getTraceId(), span.getParentSpanId()));
+final Future<HttpResponse> responseFuture = client.execute(request, null, null);
+final HttpResponse response = responseFuture.get();
+// process response here
+```
+
+Example creating request with Honeycomb Trace set. This example uses a callback for processing.
+```java
+final HttpGet request = new HttpGet("http://google.com");
+request.addHeader("x-honeycomb-trace", MessageFormat.format("1;trace_id={0},parent_id={1}", span.getTraceId(), span.getParentSpanId()));
```
Comment (paulosman) [inline beeline-core/src/main/java/io/honeycomb/beeline/tracing/README.md:52]: Isn't this a constant in `HttpHeaderV1PropagationCodec`? These examples require the user to know the format of the header - is there a way to encode the context using that codec class?

File: beeline-core/src/main/java/io/honeycomb/beeline/tracing/README.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+```java
+// `environment` is an instance of io.dropwizard.setup.Environment
+// jerseyConfiguration is an instance of io.dropwizard.client.JerseyClientConfiguration
+JerseyClientBuilder builder = new io.dropwizard.client.JerseyClientBuilder(environment);
+client = builder.using(jerseyConfiguration).build("example");
+// Once client is created, use JAX-RS API as normal
+final WebTarget target = client.target("https://google.com").path("/maps");
+final Response response = target.request(MediaType.APPLICATION_JSON_TYPE)
+    .header("x-honeycomb-trace", MessageFormat.format("1;trace_id={0},parent_id={1}", span.getTraceId(), span.getParentSpanId()))
+    .get();
```
Comment (paulosman) [inline beeline-core/src/main/java/io/honeycomb/beeline/tracing/README.md:143]: Love the examples, and the range of clients, but I'm wondering if we can simplify using available API constants methods to generate the header.

Comment (MarsianMan) [reply beeline-core/src/main/java/io/honeycomb/beeline/tracing/README.md:143]: Updated to use constant instead of hardcoded `"x-honeycomb-trace"`

Comment (paulosman) [reply beeline-core/src/main/java/io/honeycomb/beeline/tracing/README.md:143]: Better, but still requires the user to know the internal format of the wire protocol (including the version, etc). It also doesn't show off all the features of the header. I know it's more code, but I was thinking of something like:

```
HttpHeaderV1PropagationCodec codec = HttpHeaderV1PropagationCodec.getInstance();
Map<String, Object> context = Stream.of(new Object[][] {
            { "user_id", 1123 },
            { "user_name", "alyssaphacker" },
}).collect(Collectors.toMap(data -> (String) data[0], data -> data[1]));
Optional<String> headerValue = codec.encode(
    new PropagationContext(span.getTraceId(), span.getParentSpanId(), "dataset", context));
final Response response = target.request(MediaType.APPLICATION_JSON_TYPE)
    .header(HttpHeaderV1PropagationCodec.HONEYCOMB_TRACE_HEADER, headerValue.orElse(""))
    ....

```

Comment (paulosman) [reply beeline-core/src/main/java/io/honeycomb/beeline/tracing/README.md:143]: You could simplify, or make it more idiomatic, but it demonstrates:

* Sending dataset, context fields
* Using encode to safely convert a PropagationContext into a header value

Without requiring the caller to know the format of our trace header.

Comment (MarsianMan) [reply beeline-core/src/main/java/io/honeycomb/beeline/tracing/README.md:159]: I have  pulled the implementation portion out of each example and put it into a Reference utility function. I believe this enhances the conciseness and readability.

**Pull request merged**